include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -931,7 +935,16 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
                 mThreadParams.initFromConfig(*m_help, m_id, outputDirectory);
 
                 /* window selection */
-                mThreadParams.window = MovingWindow::getInstance().getWindow(currentStep);
+                auto simulationOutputWindow = MovingWindow::getInstance().getWindow(currentStep);
+
+                // set default if the user is not providing the parameter.
+                std::string selectedRange = ":,:,:";
+                if(m_help->range.optionDefined(m_id) && !m_help->range.get(m_id).empty())
+                    selectedRange = m_help->range.get(m_id);
```

Comment:
Could this be done not here, in the computational function, but somewhere before that?

Reply:
(I mean, for clarity reasons, and so that this part is not directly dependent on command-line argument)

Reply:
I checked the code. I like to avoid that we have `selectedRange` as a global attribute for an IO instance.
In case we allow defining more than one range per IO instance in the future the current location of this code is fine.